components/eamxx/src/physics/rrtmgp/eamxx_rrtmgp_process_interface.cpp

```diff
@@ -565,6 +565,50 @@ void RRTMGPRadiation::run_impl (const double dt) {
     shr_orb_decl_c2f(calday, eccen, mvelpp, lambm0,
                      obliqr, &delta, &eccf);
 
+    // Precompute VMR for all gases, on all cols, before starting the chunks loop
+    const auto gas_mol_weights = m_gas_mol_weights;
+    for (int igas = 0; igas < m_ngas; igas++) {
+      auto name = m_gas_names[igas];
+
+      // We read o3 in as a vmr already
+      if (name=="o3") continue;
+
+      auto d_vmr = get_field_out(name + "_volume_mix_ratio").get_view<Real**>();
+      if (name == "h2o") {
+        // h2o is (wet) mass mixing ratio in FM, otherwise known as "qv", which we've already read in above
+        // Convert to vmr
+        const auto policy = ekat::ExeSpaceUtils<ExeSpace>::get_default_team_policy(m_ncol, m_nlay);
+        Kokkos::parallel_for(policy, KOKKOS_LAMBDA(const MemberType& team) {
+          const int icol = team.league_rank();
+          Kokkos::parallel_for(Kokkos::TeamVectorRange(team, nlay), [&] (const int& k) {
+            d_vmr(icol,k) = PF::calculate_vmr_from_mmr(gas_mol_weights[igas],d_qv(icol,k),d_qv(icol,k));
+          });
+        });
+        Kokkos::fence();
+      } else if (name == "n2") {
```

Comment:
These two could probably be done during init. The perf gain may not be noticeable anyways though.